webapp/channels/src/components/permalink_view/actions.ts

```diff
@@ -88,25 +89,48 @@ export function focusPost(postId: string, returnTo = '', currentUserId: string,
         if (privateChannelJoinPromptVisible) {
             return;
         }
-        const {data} = await dispatch(getPostThread(postId));
 
-        if (!data) {
+        let postInfo;
+        try {
+            postInfo = await Client4.getPostInfo(postId);
+        } catch (e) {
             getHistory().replace(`/error?type=${ErrorPageTypes.PERMALINK_NOT_FOUND}&returnTo=${returnTo}`);
             return;
         }
 
-        if (data.first_inaccessible_post_time) {
+        const state = getState();
+        const currentTeam = getCurrentTeam(state);
+
+        if (!postInfo.has_joined_channel) {
+            // Prompt system admin before joining the private channel
+            const user = getCurrentUser(state);
+            if (postInfo.channel_type === Constants.PRIVATE_CHANNEL && isSystemAdmin(user.roles)) {
+                privateChannelJoinPromptVisible = true;
```

Comment:
No changes to make here, but I really don't like that we have this global state for this. It's from existing code so it's not something to fix in this PR though